flink-kubernetes-operator-autoscaler/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/metrics/ScalingMetrics.java

```diff
@@ -38,10 +38,12 @@ public class ScalingMetrics {
     private static final Logger LOG = LoggerFactory.getLogger(ScalingMetrics.class);
 
     /**
-     * The minimum value to avoid using zero values which cause side effects like division by zero
-     * or out of bounds (infinitive) floats.
+     * A large, effectively infinite value which continues to allow arithmetics for the scaling
+     * algorithm. It is used in case the task is not processing data or is completely idle. We can't
+     * accurately sample its true processing rate in this case and assume a high capacity until we
+     * know any better. This is to allow scale down in case a task becomes completely idle.
      */
-    public static final double EFFECTIVELY_ZERO = 1e-10;
+    public static final double EFFECTIVELY_INFINITE = 1e100;
```

Comment:
Any reason for not actually using `Double.POSITIVE_INFINITY`?

Reply:
Updated pushed.